experimental/framework/fluid-static/src/serviceAudience.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { IAudience } from "@fluidframework/container-definitions";
+import { Container } from "@fluidframework/container-loader";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { IServiceAudience, IServiceAudienceEvents, IMember } from "./types";
+
+// Base class for providing audience information for sessions interacting with FluidContainer
+// This can be extended by different service-specific client packages to additional parameters to
+// the user and client details returned in IMember
+export class ServiceAudience extends TypedEventEmitter<IServiceAudienceEvents> implements IServiceAudience<IMember> {
+  protected readonly audience: IAudience;
+
+  constructor(
+      protected readonly container: Container,
+  ) {
+    super();
+    this.audience = container.audience;
+
+    // Consolidating both the addition/removal of members
+    this.audience.on("addMember", () => {
+      this.emit("membersChanged", this.getMembers());
+    });
+
+    this.audience.on("removeMember", () => {
+      this.emit("membersChanged", this.getMembers());
+    });
+
+    this.container.on("connected", () => {
```

Comment:
I had added the connected listener as the userId field for the current user (self) comes back as undefined until the user actually connects with the service. As such, we needed to refresh the list of members when the user connected.

For the disconnected case,
1) If it is another user who is disconnecting, it would trigger a "removeMember" event on the underlying container's audience and be emitted by `ServiceAudience` as a "membersChanged" event
2) If is the current user (self) who disconnected, they would know from the `FluidContainer`'s disconnected event

As such, in both cases, it looks like the event would be appropriately alerted to the developer?